src/wallet/bdk_wallet/wallet.rs

```diff
@@ -1,19 +1,48 @@
 //! The Wallet API.
 
-use super::Keychain;
+use super::{
+    error::{BdkError, BdkStoreError},
+    Keychain,
+};
+use crate::wallet::FidelityError;
 use bdk_chain::{
-    bitcoin::{key::Secp256k1, secp256k1::All, Network},
+    bitcoin::{
+        bip32::IntoDerivationPath, constants::genesis_block, key::Secp256k1, secp256k1::All,
+        Network,
+    },
     keychain::KeychainTxOutIndex,
     local_chain::LocalChain,
-    CombinedChangeSet, ConfirmationTimeHeightAnchor, IndexedTxGraph,
+    Append, CombinedChangeSet, ConfirmationTimeHeightAnchor, IndexedTxGraph,
+};
+use bdk_wallet::{
+    descriptor::{calc_checksum, Descriptor, DescriptorError},
+    keys::DescriptorPublicKey,
+    signer::SignersContainer,
+    wallet::LoadError,
 };
-use bdk_wallet::signer::SignersContainer;
+use bitcoin::bip32::Xpriv;
+use std::io::{Error as FileError, ErrorKind};
+
+use crate::wallet::WalletError;
+
+use std::{convert::TryFrom, fs};
 
-use crate::wallet::WalletStore as MetaStore;
+use crate::{
+    bitcoind::bitcoincore_rpc::RpcApi,
+    wallet::{RPCConfig, WalletStore as MetaStore},
+};
+use bdk_file_store::Store as BDKStore;
 use bitcoind::bitcoincore_rpc::Client;
 use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+// Type alias of `CombinedChangeSet` having `ConfirmationTimeHeightAnchor` & custom Keychain.
 pub type ChangeSet = CombinedChangeSet<Keychain, ConfirmationTimeHeightAnchor>;
 
+// TODO: find better use of MAGIC_BYTES used in bdk_store.
+// Currently they are randomly choosed.
+const MAGIC_BYTES: [u8; 12] = [98, 100, 107, 102, 115, 49, 49, 49, 49, 49, 49, 49];
```

Comment:
The magic bytes should be the unique-id for the wallet. As discussed lets use the masterkey fingerprint here. Should be handled in the init function.